embedding-calculator/src/services/facescan/plugins/insightface/insightface.py

```diff
@@ -60,55 +53,43 @@ def scaled(self, coefficient: float) -> 'InsightFaceBoundingBox':
                                       landmark=self.landmark * coefficient)
 
 
+class InsightFaceMixin:
+    _CTX_ID = ENV.GPU_IDX
+    _NMS = 0.4
+
+    def get_model_file(self, ml_model: base.MLModel):
+        if not ml_model.exists():
+            raise exceptions.ModelImportException(
+                f'Model {ml_model.name} does not exists')
+        return model_store.find_params_file(ml_model.path)
+
+
 class DetectionOnlyFaceAnalysis(FaceAnalysis):
     rec_model = None
     ga_model = None
 
-    def __init__(self, det_name):
-        try:
-            self.det_model = model_zoo.get_model(det_name)
-        except ValueError:
-            file = _get_model_file(det_name)
-            self.det_model = face_detection.FaceDetector(file, 'net3')
+    def __init__(self, file):
+        self.det_model = face_detection.FaceDetector(file, 'net3')
 
 
-class InsightFace(FaceScanner):
-    ID = 'InsightFace'
-    DETECTION_MODEL_NAME = ENV.DETECTION_MODEL
-    CALCULATION_MODEL_NAME = ENV.CALCULATION_MODEL
-    IMG_LENGTH_LIMIT = ENV.IMG_LENGTH_LIMIT
+class FaceDetector(InsightFaceMixin, base.BaseFaceDetector):
+    ml_models = (
+        ('retinaface_r50_v1', 'http://insightface.ai/files/models/retinaface_r50_v1.zip'),
```

Comment:
The same - let's store this models in our google drive
Or as an alternative - look at this solution - https://git-lfs.github.com/